### PR TITLE
⌨️ feat: Improve keyboard navigation and shortcuts for Feedback Task

### LINKED ISSUE #3264: Improve keyboard navigation for Feedback Task

### PR DESCRIPTION
**Pull request by leiyre**
# Description

Improve keyboard navigation for Feedback Task:

- [x] Make status dropdown focusable and selectable using keyboard
- [x] Make labels and rating items focusable and selectable
- [x] Simplify code for search label focus status
- [x] Show focused first option in first question in form
- [x] Navigate between  labels using `TAB` 
- [x] Select label using `SPACE`
- [x] Click button using `ENTER`
- [x] Allow navigate between questions using `Shift + ↑` and `Shift + ↓`. 
- [x] Display elements in ranking slots using `1`, `2`, `3`, `4`, `5`, `6`, `7`,`8`,`9`,`10`,`11`, `12` (12 max)

Closes #3264

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: frontend/components/commons/forms/questions-form/QuestionsForm.component.vue [hunk 1]
```diff
@@ -40,35 +50,42 @@
...
         />
-
         <RankingComponent
           v-if="input.component_type === COMPONENT_TYPE.RANKING"
           :title="input.question"
           :isRequired="input.is_required"
+          :isFocused="checkIfQuestionIsFocused(index)"
           :description="input.description"
           v-model="input.options"
+          :key="JSON.stringify(input.options)"
```
Comment (damianpumar) [inline frontend/components/commons/forms/questions-form/QuestionsForm.component.vue]: Is it necessary the options parsed as string for the shortcuts things?

File: frontend/components/commons/forms/questions-form/fields/ranking/Ranking.component.vue [hunk 2]
```diff
@@ -57,6 +66,16 @@ export default {
         }))
       );
     },
+    onChangedWithKeyboard(value, rank) {
```
Comment (damianpumar) [inline frontend/components/commons/forms/questions-form/fields/ranking/Ranking.component.vue]: What is the difference to change the options with the keyboard or the mouse here? I mean, the business logic is the same when the user change the options with keyboard or mouse

File: frontend/components/commons/forms/questions-form/QuestionsForm.component.vue (hunk 1 above)
Comment (leiyre) [reply frontend/components/commons/forms/questions-form/QuestionsForm.component.vue]: We need to force re-rendering the component after ranking using keyboard, I tried other approach without success, do you have any idea?

File: frontend/components/commons/forms/questions-form/fields/ranking/Ranking.component.vue (hunk 2 above)
Comment (leiyre) [reply frontend/components/commons/forms/questions-form/fields/ranking/Ranking.component.vue]: yes, we can use the same

File: frontend/components/commons/forms/questions-form/QuestionsForm.component.vue (hunk 1 above)
Comment (leiyre) [reply frontend/components/commons/forms/questions-form/QuestionsForm.component.vue]: I have included more logic in children and seems to work

File: frontend/components/commons/forms/questions-form/fields/shared-components/drag-and-drop-selection/DndSelection.component.vue [hunk 3]
```diff
@@ -56,11 +64,90 @@ export default {
...
+  watch: {
+    isFocused: {
+      immediate: true,
+      handler(newValue) {
+        !!newValue && this.onAutoFocusFirstItem();
+      },
+    },
   },
   methods: {
+    rankWithKeyboard(event, item) {
```
Comment (damianpumar) [inline frontend/components/commons/forms/questions-form/fields/shared-components/drag-and-drop-selection/DndSelection.component.vue]: I would like us to review this method together

File: frontend/components/commons/sidebar-feedback-task/IconButton.vue [hunk 4]
```diff
@@ -103,6 +103,13 @@ export default {
       fill: palette(grey, 100);
     }
   }
+  &.custom-expandable :deep(svg) {
+    fill: palette(purple, 200);
+    border: 1px solid palette(purple, 200);
+    border-radius: $border-radius;
+    max-height: 20px;
+    max-width: 20px;
+  }
```
Comment (keithCuniah) [inline frontend/components/commons/sidebar-feedback-task/IconButton.vue]: move logic/css for the sidebar outside of the icon component (or the icon component become less generic)

**Pull request merged**